### PR TITLE
Refactor UpdateProductSeoHandler to use ProductRepository

### LINKED ISSUE #14776: [EPIC] Product page - SEO tab 8/9
### - Here is the list of the elements of SEO tab:
[https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/catalog/catalog-products-add-edit.md#seo-tab-epic](https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/catalog/catalog-products-add-edit.md#seo-tab-epic) 

- [x] #19677 - [SEO tab] - Search Engine Optimization 1/2
- [x] #19678 - [SEO tab] - Redirection page 2/2

### [- As a merchant I want to know if there are errors when I complete a field in the product page](https://github.com/PrestaShop/prestashop-specs/blob/master/back-office/catalog/catalog-products-add-edit.md#seo-tab-)

---
## Issues in **SEO tab** sorted by priority high to low :

**Should-Have**
- [x] #14984 - SEO preview on product page
- [ ] #15936 - Friendly url not create in some case.
- [ ] #17078 - Product Meta description not filled by default in the SEO preview

## Improvement
- [ ] #14317 - Merchant should be able to choose to index/noindex each prestashop url

## Fixed
- [x] #14054 - URL in SEO preview is not correct when the product is not active
- [x] #21495 - Bug on "reset url" button of the product pages when "Accented URLs" is enabled and there are special characters in the product name

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Part of product page migration. Refactor UpdateProductSeoHandler to use extra layer of domain services instead of handling domain logic 
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | part of #14776
| How to test?  | CI :heavy_check_mark: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21345)

### CODE REVIEW

**Comment by zuk3975** [comment]
rr travis

File: src/Adapter/Product/Update/ProductSeoInfoFiller.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    /**
+     * @param Product $product
+     * @param RedirectOption $redirectOption
+     *
+     * @return string[] updatable properties
+     *
+     * @throws CategoryNotFoundException
+     * @throws CoreException
+     */
+    public function fillRedirectOption(Product $product, RedirectOption $redirectOption): array
```
Comment (matks) [inline src/Adapter/Product/Update/ProductSeoInfoFiller.php]: Suggestion:
```suggestion
    public function injectRedirectOptionIntoProduct(Product $product, RedirectOption $redirectOption): array
```

**Review by matks — DISMISSED** [review]
Good for me 👍  added 1 suggestion

File: src/Adapter/Product/Update/ProductSeoInfoFiller.php (hunk 1 above)
Comment (zuk3975) [reply src/Adapter/Product/Update/ProductSeoInfoFiller.php]: hmm sounds strange for me :thinking:

**Comment by zuk3975** [comment]
> Good for me  added 1 suggestion

So are you good with that `{...}InfoFiller` pattern?
because that was what mainly holded this and one other PR where it was introduced.

File: src/Adapter/Product/Update/ProductSeoInfoFiller.php (hunk 1 above)
Comment (zuk3975) [reply src/Adapter/Product/Update/ProductSeoInfoFiller.php]: This discussion was moved to slack. This is what we decided.
So the initial concern was the confusion related to name `it was unclear whether the Product or the RedirectOption is being filled`.
So if we rename the class to `ProductSeoPropertiesFiller` and the method `fillWithRedirectOption` it becomes much clearer!

**Review by sowbiba — DISMISSED** [review]
LGTM

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Please add unit tests

**Comment by zuk3975** [comment]
TODO - failing to to some caching related to ps_facetedsearch module.
Fixed in #21103

**EDIT** :heavy_check_mark:

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @zuk3975

**Pull request merged**